.github/workflows/bleeding-edge.yaml

```diff
@@ -0,0 +1,48 @@
+name: CI (bleeding edge)
+# this workflow is heavily inspired from pandas, see
+# https://github.com/pandas-dev/pandas/blob/master/.github/workflows/python-dev.yml
+
+# gcheck stability against dev version of Python, numpy, and matplotlib
+
+on:
+  push:
+    branches:
+      - master
+  # Run daily at 0:01 UTC
+  schedule:
+    - cron:  '1 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Python3.10-dev
+    timeout-minutes: 60
+
+    concurrency:
+      group: ${{ github.ref }}-dev
+      cancel-in-progress: true
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python Dev Version
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10-dev'
+
+    - name: Install dependencies
+      run: |
+        python3 -m pip install --upgrade pip
+        python3 -m pip install --upgrade setuptools wheel
+        python3 -m pip install git+https://github.com/numpy/numpy.git
```

Comment:
Hi @neutrinoceros @ngoldbaum. This is a belated comment but while doing some looking at the CI for PR #214 I noticed this was created and thought I'd note that we were told not do this (building NumPy or SciPy from source) in https://github.com/scikit-hep/pyhf/pull/1533 by Ralf and asked to use the nightly wheels instead. c.f https://anaconda.org/scipy-wheels-nightly/repo

Reply:
I don't understand why we'd want to take one step further away from the source since it's trivial to build numpy ourselves. The time it takes (about 20min) seems irrelevant given this job isn't run on PRs. What am I missing ?

Reply:
Because users of numpy don’t normally build it from source so it’s a more realistic test of the distribution.